framework/hpp_gui.cpp

```diff
@@ -653,7 +668,13 @@ void HPPGui::draw(vk::CommandBuffer command_buffer) const
 			command_buffer.drawIndexed(cmd->ElemCount, 1, index_offset, vertex_offset, 0);
 			index_offset += cmd->ElemCount;
 		}
+#if defined(PLATFORM__MACOS) && TARGET_OS_IOS && TARGET_OS_SIMULATOR
```

Comment:
I'm a bit torn on "polluting" code with a special case as the iOS simulator. Can this be solved in a more elegant way? I wouldn't mind if we'd simply disable the UI on the iOS simulator instead of adding specific code for it.

Reply:
I debated this myself, and actually tried the UI disable route initially.  However, I didn't like the experience, especially in batch mode where you really can't tell what is going on as the sample runs through its various configurations.  And also in the near future I would to add gesture and key input handling for iOS and without the UI present on the Simulator that would be completely inaccessible.

The reason I like the iOS Simulator is that it is easy to run and does not require an Apple Development Certificate nor a unqiue bundle identifier for iOS.  I figured this would make your project easily accessible for iOS testing without adding overhead for potential users.  In addition, the Simulator allows me to try different devices that I don't own (e.g. larger screen and various generation iOS devices).

If you don't like the **#ifdef** special case handling, the change that I made is completely generic and could be applied to all platforms, i.e. simply don't use vertex_offset > 0 and use my rebinding code for all platforms.  Not sure if you would like that but it would make the code generic.  Please let me know.

Reply:
Maybe if we combine them into a new macro that's APPLE specific.  Say SAMPLES_PLATFORM_APPLE or similar.  I think the path of having macros is fine but there are other platforms in the Apple ecosystem that we haven't added support for yet.